src/main/java/com/commercetools/sync/commons/utils/CommonTypeUpdateActionUtils.java

```diff
@@ -0,0 +1,31 @@
+package com.commercetools.sync.commons.utils;
+
+
+import com.commercetools.sync.commons.helpers.SyncResult;
+import io.sphere.sdk.commands.UpdateAction;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+public class CommonTypeUpdateActionUtils {
+
+    /**
+     * Compares two {@link Object} and returns a {@link SyncResult} containing the supplied {@link UpdateAction} as a
+     * result. If no update action is needed, for example in case where both the {@link Object}
+     * have the same values, a sync result with containing an empty list of update actions is returned.
+     *
+     * @param oldObject    the object which should be updated.
+     * @param newObject    the object with the new information.
+     * @param updateAction the update action to return in the optional.
+     * @return A sync result with a list containing the update action or a sync result containing an empty list
+     * of update actions if the object values are identical.
+     */
+    @Nonnull
+    public static <T> SyncResult<T> buildUpdateAction(@Nullable final Object oldObject,
```

Comment:
@butenkor The naming indeed could be confusing. The real intention of this wrapper was to be used for any process that happens across the sync module that could have statistics (such as warnings/errors) that need to be tracked, and that is the wrapper's benefit. For example, tracking if a parent field was empty in the change parent update action. 

Since it is, as you said, not an actual _sync_ but actually _building update action_. We could rename the wrapper `Result` or `ProcessResult` to be used globally for all process results across all utils of the module. 

What do you think?